res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,773 @@
+                "group_" + group + "_enable");
+            engine.setValue("[EffectRack1_EffectUnit2]", "group_" + group +
+                "_enable", oldvalue ? 0 : 1);
+        }
+    }
+};
+
+ReloopBeatmix24.FX1Turn = function(channel, control, value, status, group) {
+    if (FxMode == 1) {
+        engine.setParameter(group, "parameter1", script.absoluteLin(value,
+            0, 1));
+    }
+    // Nothing in multi-effect mode
+};
+
+ReloopBeatmix24.FX2Turn = function(channel, control, value, status, group) {
+    if (FxMode == 1) {
+        engine.setParameter(group, "parameter2", script.absoluteLin(value,
+            0, 1));
+    }
+    // Nothing in multi-effect mode
+};
+
+ReloopBeatmix24.FX3Turn = function(channel, control, value, status, group) {
+    if (FxMode == 1) {
+        engine.setParameter(group, "parameter3", script.absoluteLin(value,
+            0, 1));
+    }
+    // Nothing in multi-effect mode
+};
+
+ReloopBeatmix24.FX4Turn = function(channel, control, value, status, group) {
+    if (FxMode == 1) {
+        engine.setParameter(group, "parameter4", script.absoluteLin(value,
+            0, 1));
+    } else {
+        var effectUnit = parseInt(group.substr(23, 1), 10);
+        var storeIndex = "FX4U" + effectUnit.toString();
+        if (storeIndex in previousValue) {
+            if (value - previousValue[storeIndex] > 5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect1]", "next_effect", 1);
+                previousValue[storeIndex] = value;
+            } else if (value - previousValue[storeIndex] < -5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect1]", "prev_effect", 1);
+                previousValue[storeIndex] = value;
+            }
+        } else {
+            previousValue[storeIndex] = value;
+        }
+    }
+};
+
+ReloopBeatmix24.FX5Turn = function(channel, control, value, status, group) {
+    if (FxMode == 1) {
+        engine.setParameter(group, "parameter5", script.absoluteLin(value,
+            0, 1));
+    } else {
+        var effectUnit = parseInt(group.substr(23, 1), 10);
+        var storeIndex = "FX5U" + effectUnit.toString();
+        if (storeIndex in previousValue) {
+            if (value - previousValue[storeIndex] > 5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect2]", "next_effect", 1);
+                previousValue[storeIndex] = value;
+            } else if (value - previousValue[storeIndex] < -5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect2]", "prev_effect", 1);
+                previousValue[storeIndex] = value;
+            }
+        } else {
+            previousValue[storeIndex] = value;
+        }
+    }
+};
+
+ReloopBeatmix24.FX6Turn = function(channel, control, value, status, group) {
```

Comment:
I'm not sure it would be easy to combine these functions. Doing that, I need a way to know if the knob is shifted or not. The only way I can think of is looking at the "status" parameter, which contains the midino, but that would add complexity in the function with low benefits and that would need more javascript integration of the midi mapping, this goes against my will to have most of the mapping done in XML.

I'll add a comment though.